app/src/main/java/bg/zahov/app/ui/workout/start/StartWorkoutViewModel.kt

```diff
@@ -1,117 +1,229 @@
+                Workout(
+                    id = generateRandomId(),
+                    name = if (newWorkout.name.contains("duplicate")) newWorkout.name else "${newWorkout.name} duplicate",
+                    duration = 0,
+                    volume = 0.0,
+                    date = LocalDateTime.now(),
+                    isTemplate = true,
+                    exercises = newWorkout.toWorkout(exerciseTemplates).exercises
+                )
+            )
         }
     }
 
-    fun startWorkoutFromTemplate(position: Int) {
+    /**
+     * Deletes a template workout from the repository.
+     *
+     * This method deletes the workout (converted from the `StartWorkout` object) from the repository, effectively removing
+     * the template workout.
+     *
+     * @param workout The `StartWorkout` object representing the workout to delete.
+     */
+    fun deleteTemplateWorkout(workout: StartWorkout) {
         viewModelScope.launch {
-            _templates.value?.get(position)?.let {
-                workoutState.startWorkout(it)
-            }
+            repo.deleteTemplateWorkout(workout.toWorkout(exerciseTemplates))
         }
     }
+}
 
-    fun deleteTemplateWorkout(position: Int) {
-        val list = _templates.value.orEmpty().toMutableList()
-        val toBeRemoved = list.removeAt(position)
-        viewModelScope.launch {
-            toBeRemoved.let { repo.deleteTemplateWorkout(it) }
-        }
-        _templates.value = list
+/**
+ * Converts a string representing a date and time into a `LocalDateTime` object.
+ *
+ * This function takes a string formatted as "HH:mm, d MMMM" (e.g., "13:00, 5 September") and parses it into a
+ * `LocalDateTime` object for the current year. This is useful when the input string does not include the year,
+ * but it is required for creating a `LocalDateTime` object.
+ *
+ * @return A `LocalDateTime` object that corresponds to the parsed date and time, with the current year.
+ */
+fun String.toLocalDateTime(): LocalDateTime =
+    LocalDateTime.parse(
+        "${LocalDate.now().year} $this",
+        DateTimeFormatter.ofPattern("yyyy HH:mm, d MMMM")
+    )
 
-    }
+/**
+ * Converts a `StartWorkout` to a `Workout` object.
+ *
+ * This extension function maps the `StartWorkout` to a `Workout` object, converting each exercise in the workout to
+ * an `Exercise` object. The function uses a regular expression to parse exercise names and sets. It looks for a
+ * matching `Exercise` template from the provided list (`exerciseTemplates`) based on the exercise's name.
+ *
+ * @param exerciseTemplates A list of available exercise templates that are used to determine additional information
+ * such as body part and category.
+ * @return A `Workout` object created from the `StartWorkout`.
+ */
+fun StartWorkout.toWorkout(exerciseTemplates: List<Exercise>) = Workout(
+    id = this.id,
+    name = this.name,
+    duration = 0,
+    volume = 0.0,
+    date = this.date.toLocalDateTime(),
```

Comment:
Wait. In [Workout -> StartWorkout](https://github.com/HauntedMilkshake/fitness_app/pull/27/files#diff-ca1f0e57298c0163f514bc61bc2b3142d9fbc9b08cd0f09193c7760fd1380701R150) you're converting LocalDateTime to String and here you're converting it back to LocalDateTime?

Reply:
The screen allows the user to duplicate template workouts, which adds them in firebase as Workout so I had to map them somehow.

Reply:
Why not use only `Workout` model? Or use `LocalDateTime` instead of `String` in `StartWorkout`?

Reply:
changed field to localDateTime in StartWorkout